test/integration/test-amp-analytics.js

```diff
@@ -55,6 +55,54 @@ describe.configure().skipIfPropertiesObfuscated().run('amp' +
     });
   });
 
+  describes.integration('amp-analytics click trigger', {
+    body: `
+      <a href="javascript:;"
+          data-vars-foo-bar="hello world"
+          data-vars-bar-foo="2">
+        Anchor
+      </a>
+      <amp-analytics>
+        <script type="application/json">
+        {
+          "requests": {
+            "endpoint": "${RequestBank.getUrl()}"
+          },
+          "triggers": {
+            "click": {
+              "on": "click",
+              "selector": "a",
+              "request": "endpoint",
+              "extraUrlParams": {
+                "f": "\${fooBar}",
+                "b": "\${barFoo}"
+              }
+            }
+          },
+          "vars": {
+            "barFoo": 1
+          }
+        }
+        </script>
+      </amp-analytics>
+      `,
+    extensions: ['amp-analytics'],
+  }, env => {
+    let browser;
+
+    beforeEach(() => {
+      browser = new BrowserHandle(env.win);
+    });
+
+    it('should send request', () => {
+      const reqPromise = RequestBank.withdraw().then(req => {
+        expect(req.url).to.equal('/?f=hello%20world&b=2');
+      });
+      browser.click('a');
```

Comment:
Looks good. But is it possible to test that the request is sent after the click, or not if we click on other element? With fake clock maybe?

Reply:
For such an integration test fake clock might not be feasible. 
I thought about delaying the click for 1s, and then look at the req timestamp to make sure it's after the click. But I feel not worth the wait, as it's a very unlikely regression (fire a click trigger without click).

Such an approach could be used for "element visible" trigger, where such a regression is more likely to happen.